src/vellum/lib/test_suites/resources.py

```diff
@@ -0,0 +1,134 @@
+import os
+import time
+from typing import Callable
+
+from src.vellum.lib.test_suites.constants import DEFAULT_MAX_POLLING_DURATION_MS, DEFAULT_POLLING_INTERVAL_MS
+from src.vellum.lib.test_suites.exceptions import TestSuiteRunResultsException
+from src.vellum.types.named_test_case_variable_value_request import NamedTestCaseVariableValueRequest
+from src.vellum.types.test_case_variable_value import TestCaseVariableValue
+from src.vellum.client import Vellum
+from src.vellum.types import (
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(
+        self,
+        id: str,
+        polling_inteval: int = DEFAULT_POLLING_INTERVAL_MS,
+        max_polling_duration: int = DEFAULT_MAX_POLLING_DURATION_MS,
+    ) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self._state = TestSuiteRunState.QUEUED
+        self._executions: list[TestSuiteRunExecution] | None = None
+        self._polling_interval = polling_inteval
+        self._max_polling_duration = max_polling_duration
+
+    def get_metric_outputs(
+        self, metric_identifier: str | None = None, output_identifier: str | None = None
+    ) -> list[TestSuiteRunMetricOutput]:
```

Comment:
I'm having trouble deciding whether to return a flat list here, a nested list, or a list of a new object that points to test cast id and list of metric outputs. I suppose I'm having trouble visualize how this gets use. Particularly bc of [this comment](https://www.notion.so/Externally-Evaluated-Entities-Triggering-Test-Suites-as-Code-9d64e00548d94326a5fe169385038a8b?d=5ca8116b0e554c22aacf3508777011c9&pvs=4#f8a7d38a0de040868b121a07858532ac) I left